checkov/terraform/parser.py

```diff
@@ -565,7 +568,26 @@ def parse_hcl_module(
 
         return module, tf_definitions
 
+    def _remove_unused_resolved_recursive(self, resolved):
```

Comment:
@achiar99 This method is not recursive, can we rename it?
Also, please document each method to understand its purpose